torch/_inductor/codecache.py

```diff
@@ -656,7 +656,10 @@ def compile(cls, graph, source_code, cuda):
             lock_dir = get_lock_dir()
             lock = FileLock(os.path.join(lock_dir, key + ".lock"), timeout=LOCK_TIMEOUT)
             with lock:
-                output_so = f"{input_path[:-4]}.so"
+                output_so_dir = input_path[:-4]
+                if not os.path.exists(output_so_dir):
+                    os.makedirs(output_so_dir, exist_ok=True)
```

Comment:
Why do you need `exist_ok=True` after you checked `not os.path.exists(output_so_dir)`?
(`exist_ok` is only about the final target dir, not about intermediate dirs).